### PR TITLE
Updated build with 2.12.1 version

### LINKED ISSUE #72: Add Scala 2.12 build
Currently, the library supports Scala 2.10 and 2.11 in cross build. There are already pre-release versions of Scala 2.12 (the most recent being ~~`2.12.0-M3`~~ ~~`2.12.0-M4`~~ ~~`2.12.0-RC2`~~ `2.12.0`), but the testing dependencies do not have a 2.12 release.

Either the library needs to a) be built without tests and remove logging or b) wait for necessary libraries to be released for 2.12 first.

Putting this in the backlog until we need Scala 2.12 support. Scala IDE folks have mentioned the need to build against 2.12, but I'll wait until that discussion moves a little further along.

Dependencies:
- [x] ScalaTest (org.scalatest.scalatest)
  - Has a 2.12.0-RC2 release (for 3.0.0)
- [x] ScalaMock (https://github.com/paulbutcher/ScalaMock/issues/119)
- [x] Scala Macros (org.scalamacros.paradise and org.scalamacros.quasiquotes)
  - Has a 2.12.0-RC2 release for `org.scalamacros.paradise`
  - `org.scalamacros.quasiquotes` is only needed separately for 2.10; so, no issues with 2.12
- [x] Scala Reflect (org.scala-lang.scala-reflect)
  - Has a 2.12.0-RC2 release

In-progress sdb language:
- [x] Parboiled 2 (org.parboiled.parboiled)
  - Has a 2.12.0-RC2 release

In-progress sdb interface:
- [x] Ammonite REPL (com.lihaoyi.ammonite-repl)
- [x] Ammonite Terminal (com.lihaoyi.ammonite-terminal)

### PR DESCRIPTION
**Pull request by chipsenkbeil**
Resolves #72.

Currently, I have not tested if this compiles or if the unit/integration tests pass. Just getting the process started.

@fommil, @rorygraves, @dickwall 

[EDIT]

Based on the first run, I need to resolve the following dependency issues:
- Update to acyclic plugin from 0.1.4 (used) to 0.1.5 (supports 2.12)
- Update to ammonite dependencies from 0.7.7 (used) to COMMIT-7d7b824 (supports 2.12)
    - Updated to use new commit for 2.12.1
- Update parboiled dependency from 2.1.0 (used) to 2.1.3 (supports 2.12)

### CODE REVIEW

**Comment by fommil** [comment]
@sugakandrey

**Comment by chipsenkbeil** [comment]
Appveyor is still failing with the

```
'sbt' is not recognized as an internal or external command,
operable program or batch file.
```

I still need to copy over your new appveyor file, @fommil.

**Comment by fommil** [comment]
I think the ensime-server one is the most current

**Comment by chipsenkbeil** [comment]
Latest status is that the sources now compile. Tests do not due to me forcing errors for warnings and encountering

```
Eta-expansion of zero-argument method values is deprecated. Did you intend to write stubLocation.lineNumber()?

(stubLocation.lineNumber: Function0[Int]).when().throws(new Throwable)
                      ^
```

There's a total of 19 of these errors. Going to stop here for now (have other things to do). ScalaMock struggles with overloaded methods and this was my way of specifying the zero-argument method to mock.

File: project/Common.scala [hunk 1]
```diff
@@ -27,11 +27,16 @@ object Common {
     crossScalaVersions := Seq("2.10.6", "2.11.8", "2.12.0"),
 
     scalacOptions ++= Seq(
-      "-encoding", "UTF-8", "-target:jvm-1.6",
+      "-encoding", "UTF-8",
       "-deprecation", "-unchecked", "-feature",
       "-Xfatal-warnings"
     ) ++ (CrossVersion.partialVersion(scalaVersion.value) match {
-      case Some((2, scalaMajor)) if scalaMajor == 10 => Seq("-Ywarn-all")
+      case Some((2, scalaMajor)) if scalaMajor == 10 => Seq(
```
Comment (fommil) [inline project/Common.scala]: why not write `Some((2, 10))`?

Comment (chipsenkbeil) [inline project/Common.scala]: I copied that segment from somewhere else originally. If that works, I should just do that.

**Comment by chipsenkbeil** [comment]
@fommil, ensime server doesn't have a 2.12 build on drone yet (obviously). Looks like the ensime docker image (2.x) is not using JDK8? Looks like I'll need to add that to my debugger image or something.

Status report of 2.12:
- 6 unit test failures (all tied to a single spec... not sure if it's a bug in 2.12, scalatest, or my own code)
- 6 integration test failures (don't think some - if any - are transient this time)

Not bad results considering the amount of tests 2.12 was running against.

**Comment by fommil** [comment]
Add a file .java-version for jenv to use the jdk that you expect.

You might want to use my sbt-sensible plugin to set a bunch of build defaults.

**Comment by chipsenkbeil** [comment]
Looks like there is a bug in Scalatest for 2.12 that throws an exception when using the `whenReady` method on a future: https://github.com/scalatest/scalatest/issues/1025

I think that's the only issue with the unit tests. Integration tests have some real issues to be solved on this side, I believe.

**Comment by chipsenkbeil** [comment]
@dickwall, @fommil, @rorygraves, really starting to feel the pain of the test compilation time. Appveyor runs in parallel, although only two at once for my account it seems. With Drone running one at a time, It's going to take _so_ long to test. And, if there's a transient failure, I only have the option to restart the entire build matrix on Drone.

**Comment by chipsenkbeil** [comment]
FYI, Scala 2.12 has compiled and executed successfully on my local box and Drone. I just added a build matrix for 2.10.6, 2.11.8, and 2.12.1 for Appveyor, so I'll soon see if there's an issue with 2.12 on Windows. If not, I'll merge this in.

**Comment by chipsenkbeil** [comment]
Passed on 2.10 and 2.12 for Windows, but transient failure on 2.11. Going to merge this in now just to get the ball rolling.

**Pull request merged**